src/lib.rs

```diff
@@ -108,6 +108,7 @@ impl RandomXCache {
             let key_ptr = key.as_bytes().as_ptr() as *mut c_void;
```

Comment:
Needs a check to ensure length of key is larger than zero.

Reply:
good catch